### PR TITLE
[feature](Nereids): InferPredicates support In

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

InferPredicates support `In` (all children is literal)

such as `a = b & a in [1, 2]` -> `b in [1, 2]`

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/InferPredicatesTest.java [hunk 1]
```diff
@@ -126,18 +126,17 @@ public void inferPredicatesTest03() {
                 .matches(
                     logicalProject(
                         logicalJoin(
-                            logicalFilter(
-                                    logicalOlapScan()
-                            ).when(filter -> !ExpressionUtils.isInferred(filter.getPredicate())
+                            logicalFilter(logicalOlapScan()).when(filter -> !ExpressionUtils.isInferred(filter.getPredicate())
                                     & filter.getPredicate().toSql().contains("id IN (1, 2, 3)")),
-                            logicalOlapScan()
+                            logicalFilter(logicalOlapScan()).when(filter -> ExpressionUtils.isInferred(filter.getPredicate())
```
Comment (englefly) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/InferPredicatesTest.java:131]: A in (...) and A=B => B in (...)
this rule should be applied only when A and B are not in the same table.

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**